> @halx99 I'm not quite sure why the iOS and OSX tests are failing, unless it's something I'm not seeing in the logs.

here: /Users/runner/work/axmol/axmol/core/audio/AudioPlayer.h:78:5: error: unknown type name 'Vec3'
   78 |     Vec3 _sourcePosition; 
   
   seems you miss include "math/Vec3.h" in `AudioPlayer.h`

--- a/core/audio/AudioEngine.h
+++ b/core/audio/AudioEngine.h
@@ -26,7 +26,7 @@
 
 #pragma once
 
-#include "core/math/Vec3.h"
+#include "math/Vec3.h"
 #include "platform/PlatformConfig.h"
 #include "platform/PlatformMacros.h"
 #include "audio/AudioMacros.h"

--- a/core/audio/AudioEngineImpl.h
+++ b/core/audio/AudioEngineImpl.h
@@ -36,7 +36,7 @@
 #    include "audio/AudioMacros.h"
 #    include "audio/AudioCache.h"
 #    include "audio/AudioPlayer.h"
-#    include "core/math/Vec3.h"
+#    include "math/Vec3.h"
 
 namespace ax
 {

--- a/core/audio/AudioPlayer.h
+++ b/core/audio/AudioPlayer.h
@@ -37,6 +37,7 @@
 #include "audio/AudioMacros.h"
 #include "platform/PlatformMacros.h"
 #include "audio/alconfig.h"
+#include "math/Vec3.h"
 
 namespace ax
 {

